rsp_vision/save/save_data.py

```diff
@@ -0,0 +1,378 @@
+        The FolderNamingSpecs object.
+    config_file : dict
+        The config file.
+    """
+    #  save in a file details about the experimental data that was analysed
+    metadata = {
+        "folder_name": folder_naming_specs.folder_name,
+        "photon_type": "two_photon"
+        if photon_data.photon_type == PhotonType.TWO_PHOTON
+        else "three_photon",
+        "date": str(datetime.datetime.now()),
+        "rsp_vision_latest_commit_hash": str(
+            git.Repo(search_parent_directories=True).head.object.hexsha
+        ),
+        "config_file": config_file,
+    }
+
+    with open(session_folder.ses_folder_path / "metadata.yml", "w") as f:
+        yaml.dump(metadata, f)
+
+
+def save_info_in_main_log(
+    folder_naming_specs: FolderNamingSpecs,
+    subject_folder: SubjectFolder,
+    session_folder: SessionFolder,
+    photon_data: PhotonData,
+    reanalysis: bool,
+    analysis_log: pd.DataFrame,
+    swc_blueprint_spec: SWC_Blueprint_Spec,
+):
+    """
+    Save info about this analysis in the main log file.
+
+    Parameters
+    ----------
+    folder_naming_specs : FolderNamingSpecs
+        The FolderNamingSpecs object.
+    subject_folder : SubjectFolder
+        The SubjectFolder object.
+    session_folder : SessionFolder
+        The SessionFolder object.
+    photon_data : PhotonData
+        The PhotonData object.
+    reanalysis : bool
+        Whether this is a reanalysis.
+    analysis_log : pd.DataFrame
+        The analysis log.
+    swc_blueprint_spec : SWC_Blueprint_Spec
+        The SWC_Blueprint_Spec object.
+    """
+    dict = {
+        "folder name": folder_naming_specs.folder_name,
+        "sub": subject_folder.sub_num,
+        "ses": session_folder.ses_num,
+        "mouse line": folder_naming_specs.mouse_line,
+        "mouse id": folder_naming_specs.mouse_id,
+        "hemisphere": folder_naming_specs.hemisphere,
+        "brain region": folder_naming_specs.brain_region,
+        "monitor position": session_folder.monitor,
+        "fov": folder_naming_specs.fov if folder_naming_specs.fov else "",
+        "cre": folder_naming_specs.cre if folder_naming_specs.cre else "",
+        "analysed": True,
+        "analysis date": str(datetime.datetime.now()),
+        "commit hash": str(
+            git.Repo(search_parent_directories=True).head.object.hexsha
+        ),
+        "microscope": "two photon"
+        if photon_data.photon_type == PhotonType.TWO_PHOTON
+        else "three photon",
+        "n roi": photon_data.n_roi,
+        "n responsive roi": len(photon_data.responsive_rois),
+        "days of the experiment": photon_data.total_n_days,
+    }
+
+    if not reanalysis:
+        analysis_log = pd.concat([analysis_log, pd.DataFrame(dict, index=[0])])
+    else:
+        analysis_log.loc[
+            analysis_log["folder name"] == folder_naming_specs.folder_name
+        ] = dict
+
+    analysis_log.to_csv(swc_blueprint_spec.path / "analysis_log.csv")
```

Comment:
If you want to be super compliant to SWC-Blueprint, use tab-separated files istead of csv, see [here](https://swc-blueprint.neuroinformatics.dev/specification.html#metadata-conventions)
```suggestions
    analysis_log.to_csv(swc_blueprint_spec.path / "analysis_log.tsv", sep="\t")
```

Reply:
If you do this, remember to change csv->tsv elsewhere as well, e.g. in docstrings

Reply:
OK, I'll think about it and make a decision on csv vs tsv. Thanks

Reply:
I will keep csv for now